swift_scality_backend/diskfile.py

```diff
@@ -160,30 +171,40 @@ def _do_http(self, caller_name, handlers, method, path, headers=None):
         '''
 
         address, port = self.sproxyd_hosts.next()
+        pool = self.http_pools.connection_from_host(address, port)
         safe_path = self.base_path + urllib.quote(path)
 
-        conn = None
-
         def unexpected_http_status(response):
-            message = response.read()
+            message = response.data
 
             raise SproxydHTTPException(
                 '%s: %s' % (caller_name, message),
                 ipaddr=address, port=port,
-                path=self.base_path,
+                path=safe_path,
                 http_status=response.status,
                 http_reason=response.reason)
 
-        with swift.common.exceptions.ConnectionTimeout(self.conn_timeout):
-            conn = swift.common.bufferedhttp.http_connect_raw(
-                address, port, method, safe_path, headers)
+        response = pool.request(method, safe_path, headers=headers, preload_content=False)
 
-        with contextlib.closing(conn), eventlet.Timeout(self.proxy_timeout):
-            resp = conn.getresponse()
-            status = resp.status
+        self.logger.debug('The HTTP connection pool to %s:%d serviced %d '
```

Comment:
Not sure we should leave this in. It logs quite some data (but not necessarily highly useful data) for every request.